components/humantask/org.wso2.carbon.humantask/src/main/java/org/wso2/carbon/humantask/core/engine/runtime/xpath/JaxpFunctionResolver.java

```diff
@@ -590,11 +724,695 @@ private Node createOrgEntity(GenericHumanRoleDAO ghr) {
+                    if (state.equalsIgnoreCase(subTask.getStatus().name())) {
+                        count++;
+                    }
+                }
+            } else {
+                //invalid argument
+                throw new HumanTaskRuntimeException(
+                        "Invalid argument : " + arg + ", for function getCountOfSubTasksInState()");
+            }
+
+            return count;
+        }
+    }
+
+    //check the validity of a given task status
+    private boolean isValidStatus(String status) {
+        for (TaskStatus taskStatus : TaskStatus.values()) {
+            if (status.equalsIgnoreCase(taskStatus.name())) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns the concatenation of all string nodes,
+     * returns an empty string for an empty node-set
+     * logic: goes through the list and concat textContent to result string,
+     * return empty string for empty list.
+     * In : node-set of string nodes
+     */
+    public class Concat implements XPathFunction {
+
+        @Override
+        public Object evaluate(List args) throws XPathFunctionException {
+            String result = "";
+            if (args.size() == 1 && args.get(0) instanceof ArrayList) {
+                ArrayList nodeList = (ArrayList) args.get(0);
+                for (int i = 0; i < nodeList.size(); i++) {
+                    try {
+                        String item = (((Element) nodeList.get(i))).getTextContent();
+                        result = result.concat(item);
+                    } catch (DOMException e) {
+                        throw new HumanTaskRuntimeException(
+                                "Invalid arguments:" + args.get(0) + ", for function concat()", e);
+                    } catch (ClassCastException e) {
+                        throw new HumanTaskRuntimeException(
+                                "Invalid arguments:" + args.get(0) + " , for function concat()",e);
+                    }
+                }
+            } else {
+                throw new HumanTaskRuntimeException("Invalid arguments:" + args + " , for function concat()");
+            }
+            return result;
+        }
+    }
+
+    /**
+     * Returns the concatenation of all string nodes, separated by the specified delimiter string,
+     * returns an empty string for an empty node-set.
+     * logic: go through the list and conact textContent and the delimiter to result,
+     * if the last item of the list, delimiter not added.
+     * In : node-set of string nodes,delimiter string
+     */
+    public class ConcatWithDelimiter implements XPathFunction {
+
+        @Override
+        public Object evaluate(List args) throws XPathFunctionException {
+            String result = "";
+            if (args.size() == 2 && args.get(0) instanceof ArrayList && args.get(1) instanceof String) {
+                ArrayList nodeList = (ArrayList) args.get(0);
```

Comment:
Use proper typing associated with the declared Lists, which will then help you avoid explicitly casting things around.